server/service/src/sync/sync_buffer.rs

```diff
@@ -62,7 +63,11 @@ impl<'a> SyncBuffer<'a> {
                 SyncBufferFilter::new()
                     .table_name(EqualFilter::equal_to(legacy_table_name))
                     .action(action.equal_to())
-                    .integration_datetime(DatetimeFilter::is_null(true)),
+                    .integration_datetime(DatetimeFilter::is_null(true))
+                    .source_site_id(match source_site_id {
+                        Some(site_id) => EqualFilter::equal_to_i32(site_id),
+                        None => EqualFilter::i32_is_null(true),
```

Comment:
Is there a use case for wanting to integrate all pending sync buffer rows? Because now we can only integrate EITHER for this site (null) or for a specific site id...

Reply:
I don't think there is